src/spatialdata/_core/centroids.py

```diff
@@ -0,0 +1,146 @@
+from spatialdata.transformations.operations import get_transformation
+from spatialdata.transformations.transformations import BaseTransformation
+
+BoundingBoxDescription = dict[str, tuple[float, float]]
+
+
+def _validate_coordinate_system(e: SpatialElement, coordinate_system: str) -> None:
+    d = get_transformation(e, get_all=True)
+    assert isinstance(d, dict)
+    assert coordinate_system in d, (
+        f"No transformation to coordinate system {coordinate_system} is available for the given element.\n"
+        f"Available coordinate systems: {list(d.keys())}"
+    )
+
+
+@singledispatch
+def get_centroids(
+    e: SpatialElement,
+    coordinate_system: str = "global",
+) -> DaskDataFrame:
+    """
+    Get the centroids of the geometries contained in a SpatialElement, as a new Points element.
+
+    Parameters
+    ----------
+    e
+        The SpatialElement. Only points, shapes (circles, polygons and multipolygons) and labels are supported.
+    coordinate_system
+        The coordinate system in which the centroids are computed.
+
+    Notes
+    -----
+    For :class:`~shapely.Multipolygon`s, the centroids are the average of the centroids of the polygons that constitute
+    each :class:`~shapely.Multipolygon`.
+    """
+    raise ValueError(f"The object type {type(e)} is not supported.")
+
+
+def _get_centroids_for_axis(xdata: xr.DataArray, axis: str) -> pd.DataFrame:
+    """
+    Compute the component "axis" of the centroid of each label as a weighted average of the xarray coordinates.
+
+    Parameters
+    ----------
+    xdata
+        The xarray DataArray containing the labels.
+    axis
+        The axis for which the centroids are computed.
+
+    Returns
+    -------
+    pd.DataFrame
+        A DataFrame containing one column, named after "axis", with the centroids of the labels along that axis.
+        The index of the DataFrame is the collection of label values, sorted ascendingly.
+    """
+    centroids: dict[int, float] = defaultdict(float)
+    for i in xdata[axis]:
+        portion = xdata.sel(**{axis: i}).data
+        u = da.unique(portion, return_counts=True)
+        labels_values = u[0].compute()
+        counts = u[1].compute()
+        for j in range(len(labels_values)):
+            label_value = labels_values[j]
+            count = counts[j]
+            centroids[label_value] += count * i.values.item()
+
+    all_labels_values, all_labels_counts = da.unique(xdata.data, return_counts=True)
+    all_labels = dict(zip(all_labels_values.compute(), all_labels_counts.compute()))
+    for label_value in centroids:
+        centroids[label_value] /= all_labels[label_value]
+    centroids = dict(sorted(centroids.items(), key=lambda x: x[0]))
+    return pd.DataFrame({axis: centroids.values()}, index=list(centroids.keys()))
+
+
+@get_centroids.register(SpatialImage)
+@get_centroids.register(MultiscaleSpatialImage)
+def _(
```

Comment:
For the dispatch methods, I think it would be nice if we added a comment/docstring for each dispatch method to state which element they are for. I know the types are registered, but not everyone will know that points are in a Dask DataFrame (for example).